csrc/scheduler/normalization_inner_outer_utils.cpp

```diff
@@ -323,5 +322,49 @@ PersistentBufferStorageParams getPersistentBufferStorageParams(
   return buffer_params;
 }
 
+std::vector<TensorView*> getGroupedReductionPersistentTvs(
+    Fusion* fusion,
+    TensorView* inner_bcast_tv,
+    const std::vector<TensorView*>& reduction_tvs) {
+  std::vector<TensorView*> res;
+  // Get all fusion outputs that are consumers of reduction tvs
+  const auto& reduction_to_output = DependencyCheck::getAllOutputsOf(
+      {reduction_tvs.begin(), reduction_tvs.end()});
+  std::unordered_set<TensorView*> p_of_reductions;
+  std::unordered_set<TensorView*> c_of_reductions;
+  for (auto output : reduction_to_output) {
+    auto chains_to_output =
+        DependencyCheck::getAllDependencyChains(inner_bcast_tv, output);
+    for (auto chain : chains_to_output) {
+      auto tv_chain = ir_utils::filterByType<TensorView>(chain);
+      bool is_reduction_chain =
+          std::any_of(tv_chain.begin(), tv_chain.end(), [](TensorView* tv) {
+            return tv->hasReduction();
+          });
+      if (is_reduction_chain) {
+        for (auto tv : tv_chain) {
+          // Don't include tvs pass reduction since we only want to find tvs
+          // inlined before reduction.
+          if (tv->hasReduction()) {
+            break;
+          }
+          p_of_reductions.insert(tv);
+        }
```

Comment:
nitpick:
```suggestion
          std::copy_if(
          tv_chain.begin(),
          tv_chain.end(),
          std::inserter(p_of_reductions, p_of_reductions.begin()),
          [](TensorView* tv) {
            return tv->hasReduction();
          });
```

Reply:
This changes the logic. For example, assuming the chain is `1 -> 2 -> r3 -> 4`, the original logic generates `p_of_reductions = {1, 2}`, it stops at `r3`. The revised version generates `p_of_reductions = {1, 2, 4}`

Reply:
ah, sorry I didn't notice the break.

(also messed up the condition, I was trying to produce {1, 2, 4}. :face_exhaling: )